### ISSUE TITLE
session_timezone is not reflected in X-ClickHouse-Timezone

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#63359]
```sql
(curl -v "http://localhost:8123/?query=select+1+settings+session_timezone='Europe/Berlin'") 2>&1 |grep Timezone
< X-ClickHouse-Timezone: Etc/UTC
```

expected: X-ClickHouse-Timezone: Europe/Berlin

**Comment by zvonand** [comment]
This header was indroduced before 'session_timezone', and is not related

**Comment by den-crane** [comment]
>This header was indroduced before 'session_timezone', and is not related

I have no words. I am gonna kill myself.

**Comment by den-crane** [comment]
@zvonand read this https://github.com/ClickHouse/clickhouse-connect/issues/138#issuecomment-1472350299

Also we discussed it in the slack and Filimonov told us to add a test for X-ClickHouse-Timezone

**Comment by zvonand** [comment]
> I have no more words. I am gonna kill myself.

Why? There is a parameter in config.xml called 'timezone'. The header that is called 'timezone' reflects a parameter 'timezone'. It is not called X-Clickhouse-Session-timezone

**Comment by zvonand** [comment]
> @zvonand read this https://github.com/ClickHouse/clickhouse-connect/issues/138#issuecomment-1472350299

Well, we keep these two things separately: timezone of a server and timezone of a session. Maybe we can introduce a different header that will reflect session_timezone?

**Comment by den-crane** [comment]
@zvonand Really? Guess why Clickhouse sends `X-ClickHouse-Timezone`, what is the purpose of this header? Why clients need it?

**Comment by zvonand** [comment]
Fair enough. I'll do my best to fix it next week

**Comment by zvonand** [comment]
Yes, it is a bug. This header returned proper timezone in case the setting was applied session-wide, but failed to return the correct one when used as a query setting.
@den-crane thanks for pointing it out
